### PR TITLE
feat(spans): Add span tags endpoint

### PR DESCRIPTION
**Pull request by Zylphrex**
To support autocompletion better in the trace explorer, we need to return the span tags and autcomplete the values.

### CODE REVIEW

File: src/sentry/api/endpoints/organization_spans_tags.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+            # are returned if the total exceeds the limit.
+            builder = SpansIndexedQueryBuilder(
+                Dataset.SpansIndexed,
+                params=cast(ParamsType, params),
+                snuba_params=snuba_params,
+                query=None,
+                selected_columns=["array_join(tags.key)"],
+                # The orderby is intentionally `None` here as this query is much faster
+                # if we let Clickhouse decide which order to return the results in.
+                # This also means we cannot order by any columns or paginate.
```
Comment (k-fish) [inline src/sentry/api/endpoints/organization_spans_tags.py:57]: I don't think we need this warning all the time

File: src/sentry/api/endpoints/organization_spans_tags.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+                orderby=None,
+                limitby=("array_join(tags.key)", 1),
+                limit=max_span_tags,
+                sample_rate=options.get("performance.spans-tags-key.sample-rate"),
```
Comment (k-fish) [inline src/sentry/api/endpoints/organization_spans_tags.py:61]: I don't think we'll ever need / want this sampled, we would prob want a materialized view instead in the long run or to control via period

File: src/sentry/api/endpoints/organization_spans_tags.py [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+        paginator = SequencePaginator(
+            [
+                (
+                    row[key],
+                    TagValue(
+                        key=key,
+                        value=row[key],
+                        times_seen=row["count()"],
+                        first_seen=row["min(timestamp)"],
+                        last_seen=row["max(timestamp)"],
```
Comment (k-fish) [inline src/sentry/api/endpoints/organization_spans_tags.py:138]: TIL we get back `*seen` on tags

**Review by k-fish — APPROVED** [review]
Looks good

File: src/sentry/api/endpoints/organization_spans_tags.py (hunk 2 above)
Comment (Zylphrex) [reply src/sentry/api/endpoints/organization_spans_tags.py:61]: This is more of a stop gap solution to reduce the bytes scanned until we get there. Ive defaulted this to 1 so it doesn't do anything yet.

**Pull request merged**